jazz-build-module/scm-module.groovy

```diff
@@ -0,0 +1,179 @@
+def createProject(repo_owner, repo_name){
+    try { 
+        echo "ready to create project"
+        if(scm_config.JAZZ_SCM == "gitlab") {
+            def git_username
+            def cas_proj_id
+            def cas_repo_id
+            
+            //assuming usernames for now are a concatenation of the user's email, check platform_usermanagement index.js
+            if(repo_owner.contains("@")){
+                git_username = repo_owner.substring(0, owner.indexOf("@"))
+            }
+
+            def user_id = getGitlabUserId(git_username, ${scm_config.GITLAB.gitlab_private_token})
+            
+            def gitlab_repo_output = sh (
+                script: "curl --header \"Private-Token: ${scm_config.GITLAB.gitlab_private_token}\" -X POST \"http://${scm_config.REPOSITORY.REPO_BASE}/api/v3/projects/user/$user_id?name=$repo_name&path=$repo_name&visibility=private&request_access_enabled=true\"",
+                returnStdout: true
+            ).trim()
+
+            def jsonSlurper = new JsonSlurper()
+            def object = jsonSlurper.parseText(gitlab_repo_output)
+            
+            if(object == null || object.equals("") || object.id == null || object.id.equals("")){
+                error "project creation in gitlabs failed"
+            }
+             
+            cas_proj_id = object.id
+
+            cas_repo_id = getCasRepoId(gitlab_private_token, repo_loc)
+
+            transferProject(cas_repo_id, cas_proj_id)
+        }
+        else{
+            sh "curl -X POST -k -v -u \"${scm_config.BITBUCKET.BITBUCKET_USERNAME}:${scm_config.BITBUCKET.BITBUCKET_PASSWORD}\" -H \"Content-Type: application/json\" " + scm_user_services_api_endpoint + " -d \'{\"name\":\""+ repo_name +"\", \"scmId\": \"git\", \"forkable\": \"true\"}\'"
+        }
+    }catch (ex) {
+        if(!((ex.getMessage()).indexOf("groovy.json.internal.LazyMap") > -1)) {
+            echo "createUserInSCM Failed"
+            error "createProjectInSCM Failed. "+ex.getMessage()
+        }
+    }
+}
+
+def getGitlabUserId(gitlab_username){
+    try{
+        def output = sh (
+            script: "curl --header \"Private-Token: ${scm_config.GITLAB.gitlab_private_token}\" -X GET \"http://${scm_config.REPOSITORY.REPO_BASE}/api/v3/users?username=$gitlab_username\"",
+            returnStdout: true
+        ).trim()
+  
+        def jsonSlurper = new groovy.json.JsonSlurper()
+        def userObject = jsonSlurper.parseText(output)
+  
+        if(userObject == null || userObject.equals("") || userObject[0] == null || userObject[0].equals("") 
+            || userObject[0].id == null || userObject[0].id.equals("")){
+                error "get user data in gitlab failed"
+        }
+        
+        return userObject[0].id
+    }
+    catch (ex) {
+            if(!((ex.getMessage()).indexOf("groovy.json.internal.LazyMap") > -1)) {
+                echo "getGitlabUserId Failed"
+                error "getGitlabUserId Failed. "+ex.getMessage()
+            }
```

Comment:
Since the function returns a value, how do we handle the else part (when it is lazyMap exception)?

Reply:
Fixed it by having a generic catch as none of the exceptions are actionable.

Reply:
Fixed it by having a generic catch as none of the exceptions are actionable.